core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -155,6 +180,11 @@ public double getUtilization()
         return bufferedBytes.get() / (double) maxBufferedBytes;
     }
 
+    public TDigestHistogram getUtilizationHistogram()
```

Comment:
I don't think it's called frequently enough for synchronized `io.airlift.stats.TDigest#copyOf` to be a problem. Could you verify empirically?

Reply:
If this a problem really then you could for example just skip updating histogram via some opportunistic lock:
```
public TDigestHistogram getUtilizationHistogram() {
  recordOutputBufferUtilization();
  // get lock on bufferUtilization
  TDigestHistogram result = new TDigestHistogram(TDigest.copyOf(bufferUtilization));
  // release lock on bufferUtilization
  return result;
}

private void recordOutputBufferUtilization() {
  long recordTime = ticker.read();
  // TRY get lock on bufferUtilization
  if (lockSuccessful) {
    bufferUtilization.add(lastBufferUtilization, (double) (recordTime - this.lastBufferUtilizationRecordTime));
  }
  // release lock on bufferUtilization
  lastBufferUtilizationRecordTime = recordTime;
  lastBufferUtilization = getUtilization();
}
```

Reply:
Yes, I verify it.